.github/workflows/pr.yaml

```diff
@@ -0,0 +1,73 @@
+name: pr
+on:
+  workflow_dispatch: {}
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  unit-tests:
+    name: unit test helpers
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.18
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install --yes zlib1g-dev libelf-dev
+        shell: bash
+      - name: unit test helpers
+        run: go test ./helpers
+
+  self-test:
+    name: selftest
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.18
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install --yes zlib1g-dev libelf-dev
+        shell: bash
+      - name: selftest
+        run: |
+          sudo make selftest-static-run
+
+  compile-trace:
+    name: compile against tracee
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.18
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install --yes bsdutils
+          sudo apt-get install --yes build-essential
+          sudo apt-get install --yes pkgconf
+          sudo apt-get install --yes llvm-12 clang-12
+          sudo apt-get install --yes clang-format-12
+          sudo apt-get install --yes zlib1g-dev libelf-dev
+          for tool in "clang" "llc" "llvm-strip" "clang-format"
+          do
+            sudo rm -f /usr/bin/$tool
+            sudo ln -s /usr/bin/$tool-12 /usr/bin/$tool
+          done
```

Comment:
We should use `build-dependencies` action file here (like https://github.com/aquasecurity/tracee/blob/main/.github/actions/build-dependencies/action.yaml) IMO.

Reply:
done!